src/components/ListItem.tsx

```diff
@@ -1,8 +1,70 @@
-import * as api from '../api';
-import './ListItem.css';
+import "./ListItem.css";
+import { updateItem, ListItem } from "../api";
+import { useEffect, useState } from "react";
+import toast from "react-hot-toast";
+import { is24HoursLater } from "../utils";
 
-type Props = Pick<api.ListItem, 'name'>;
+interface Props {
+	item: ListItem;
+	listPath: string | null;
+}
+
+export function ListItemCheckBox({ item, listPath }: Props) {
+	const [checked, setChecked] = useState(false);
+
+	const purchaseDate = item.dateLastPurchased
+		? item.dateLastPurchased.toDate()
+		: item.dateLastPurchased;
+
+	useEffect(() => {
+		if (purchaseDate) {
+			setChecked(!is24HoursLater(purchaseDate));
+		} else {
+			setChecked(false);
+		}
+	}, [item.dateLastPurchased]);
+
+	const handleCheckChange = async (e: React.ChangeEvent<HTMLInputElement>) => {
+		const isChecked = e.target.checked;
+
+		if (!isChecked && purchaseDate && !is24HoursLater(purchaseDate)) {
+			toast.error(
+				`${item.name} has already been marked as purchased in the last 24 hours.`,
+			);
+			return;
+		}
+
+		setChecked(isChecked);
+
+		if (!listPath) {
```

Comment:
Going back to the idea of "making impossible states impossible", this is something that shouldn't ever be undefined here, but we're forced by the type system to contend with it being undefined. That's a "code smell" that logic elsewhere could be improved so we don't need to be handling this. Out of scope for this PR, but something to keep in mind 🙂